In [1]:
import pandas as pd

In [2]:
df=pd.read_csv('/content/drive/MyDrive/dataset_csv/GBvideos.csv', error_bad_lines=False)

/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:3326: FutureWarning: The error_bad_lines argument has been deprecated and will be removed in a future version.


  exec(code_obj, self.user_global_ns, self.user_ns)
b'Skipping line 2398: expected 11 fields, saw 21\nSkipping line 2797: expected 11 fields, saw 21\n'


In [3]:
df.head()

video_id                                              title  \
0  jt2OHQh0HoQ  Live Apple Event - Apple September Event 2017 ...   
1  AqokkXoa7uE  Holly and Phillip Meet Samantha the Sex Robot ...   
2  YPVcg45W0z4                    My DNA Test Results! I'm WHAT?!   
3  T_PuZBdT2iM  getting into a conversation in a language you ...   
4  NsjsmgmbCfc                               Baby Name Challenge!   

       channel_title  category_id  \
0        Apple Event           28   
1       This Morning           24   
2       emmablackery           24   
3              ProZD            1   
4  Sprinkleofglitter           26   

                                                tags    views  likes  \
0  apple events|apple event|iphone 8|iphone x|iph...  7426393  78240   
1  this morning|interview|holly willoughby|philli...   494203   2651   
2  emmablackery|emma blackery|emma|blackery|briti...   142819  13119   
3  skit|korean|language|conversation|esl|japanese...  1580028  65729   
4  sprinkleofglitter|sprinkle of glitter|baby gli...    40592   5019   

   dislikes  comment_total                                     thumbnail_link  \
0     13548            705  https://i.ytimg.com/vi/jt2OHQh0HoQ/default_liv...   
1      1309              0     https://i.ytimg.com/vi/AqokkXoa7uE/default.jpg   
2       151           1141     https://i.ytimg.com/vi/YPVcg45W0z4/default.jpg   
3      1529           3598     https://i.ytimg.com/vi/T_PuZBdT2iM/default.jpg   
4        57            490     https://i.ytimg.com/vi/NsjsmgmbCfc/default.jpg   

    date  
0  13.09  
1  13.09  
2  13.09  
3  13.09  
4  13.09

In [4]:
df['views'].describe()

count    7.993000e+03
mean     1.110733e+06
std      3.048740e+06
min      0.000000e+00
25%      1.083750e+05
50%      3.151940e+05
75%      9.712210e+05
max      5.896141e+07
Name: views, dtype: float64

In [5]:
df['channel_title'].value_counts()

The Tonight Show Starring Jimmy Fallon    40
First We Feast                            40
jacksfilms                                39
Nile Wilson                               38
PointlessBlogVlogs                        38
                                          ..
YandelVEVO                                 1
BD Horror Trailers and Clips               1
NBA                                        1
Guinness World Records                     1
Gladd                                      1
Name: channel_title, Length: 1003, dtype: int64

Here i want to discover how many channel are in this data set

Let see what makes Jimmy Fallon produce many videos

In [6]:
df_jimmy_fallon = df[df['channel_title'] == 'The Tonight Show Starring Jimmy Fallon']

In [7]:
df_jimmy_fallon.shape

(40, 11)

Here is 40, all good

In [8]:
df_jimmy_fallon['views'].describe().apply(lambda x: format(x, 'f'))

count         40.000000
mean     1328061.250000
std      1138216.063081
min       108480.000000
25%       588994.000000
50%       963740.500000
75%      2027072.750000
max      3988237.000000
Name: views, dtype: object

Let understand which are the video with minimum views and maximum views, in order to see which comments provoked such recordings

In [9]:
def views_less(column):

  element_less=[]
  min_col=column.min()
  counter=0

  for i in column:

    if i == min_col:

      
      print(i)

      element_less.append(counter)

    counter+=1
  
  print(element_less)

    

In [10]:
views_less(df_jimmy_fallon['views'])

108480
[10]


Let store the video with the less views

In [11]:
print(df_jimmy_fallon.iloc[10])
video_with_less_views=df_jimmy_fallon.iloc[10]

video_id                                                RgskTJ27hPY
title             Black Simon & Garfunkel Sing My Humps with Fergie
channel_title                The Tonight Show Starring Jimmy Fallon
category_id                                                      23
tags              The Tonight Show|Jimmy Fallon|Black Simon & Ga...
views                                                        108480
likes                                                          2793
dislikes                                                         77
comment_total                                                   117
thumbnail_link       https://i.ytimg.com/vi/RgskTJ27hPY/default.jpg
date                                                          23.09
Name: 2060, dtype: object


In [12]:
def views_best(column):

  element_less=[]
  max_col=column.max()
  counter=0

  for i in column:

    if i == max_col:

      
      print(i)

      element_less.append(counter)

    counter+=1
  
  print(element_less)

In [13]:
views_best(df_jimmy_fallon['views'])

3988237
[27]


In [14]:
df_jimmy_fallon.iloc[27]
video_with_more_views=df_jimmy_fallon.iloc[27]

Now let's compare with the data with comments

In [15]:
df_comments=pd.read_csv('/content/drive/MyDrive/dataset_csv/GBcomments.csv', error_bad_lines=False)

/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:3326: FutureWarning: The error_bad_lines argument has been deprecated and will be removed in a future version.


  exec(code_obj, self.user_global_ns, self.user_ns)
b'Skipping line 113225: expected 4 fields, saw 5\n'
b'Skipping line 158379: expected 4 fields, saw 7\nSkipping line 241590: expected 4 fields, saw 5\nSkipping line 245637: expected 4 fields, saw 7\n'
b'Skipping line 521402: expected 4 fields, saw 5\n'


In [16]:
df_comments

video_id                                       comment_text  likes  \
0       jt2OHQh0HoQ  It's more accurate to call it the M+ (1000) be...      0   
1       jt2OHQh0HoQ              To be there with a samsung phone\n😂😂😂      1   
2       jt2OHQh0HoQ  Thank gosh, a place I can watch it without hav...      0   
3       jt2OHQh0HoQ  What happened to the home button on the iPhone...      0   
4       jt2OHQh0HoQ  Power is the disease.  Care is the cure.  Keep...      0   
...             ...                                                ...    ...   
718447  lLN1FwiqGwc                            MEME ME MEME ME MEME ME      0   
718448  lLN1FwiqGwc                    Yeah, but does everybody float?      0   
718449  lLN1FwiqGwc                                That meme me dace 😂      0   
718450  lLN1FwiqGwc                     10:43 Is that Miles Upshur????      2   
718451  lLN1FwiqGwc                                WA-HA\nWA-HA\nWA-HA      1   

        replies  
0             0  
1             0  
2             0  
3             0  
4             0  
...         ...  
718447        0  
718448        0  
718449        0  
718450        0  
718451        0  

[718452 rows x 4 columns]

In [23]:
video_id_less_views=video_with_less_views[0]

In [24]:
df_comments_less= df_comments[df_comments['video_id']==video_id_less_views]

In [25]:
df_comments_less

video_id                                       comment_text  likes  \
181921  RgskTJ27hPY  Fergie looks so funny when she's in the moment...     75   
181922  RgskTJ27hPY                                  Fergie is so fine    191   
181923  RgskTJ27hPY                           *Beatles sing despacito*     83   
181924  RgskTJ27hPY                        Hello darkness my old nigga    251   
181925  RgskTJ27hPY                          She still looks the same!     44   
...             ...                                                ...    ...   
233880  RgskTJ27hPY  fergie did something to her lips \nshe looks l...      1   
233881  RgskTJ27hPY                 DeeZ niggas so emotional n shieeet      1   
233882  RgskTJ27hPY  But she's not in the peas anymore so she shoul...      1   
233883  RgskTJ27hPY  FERGIE??\nthe one who tried to sell a song cal...      2   
233884  RgskTJ27hPY  Excuse me, but it's not Black. It's African-Am...      1   

        replies  
181921        0  
181922        0  
181923        3  
181924        4  
181925        0  
...         ...  
233880        0  
233881        0  
233882        3  
233883        1  
233884       11  

[244 rows x 4 columns]

Usually, the comment with more likes can describe something about the video, identifying some pattern or key aspects.

In [27]:
df_comments_less['likes'].describe()

count    244.000000
mean      16.651639
std       47.236616
min        0.000000
25%        0.000000
50%        1.000000
75%        5.000000
max      287.000000
Name: likes, dtype: float64

The maximum number of likes was 287 \\
I am going to use my previous function in order to identfy the index





In [28]:
views_best(df_comments_less['likes'])

287
[180]


In [29]:
df_comments_less.iloc[180]

video_id                        RgskTJ27hPY
comment_text    Hello darkness my old nigga
likes                                   287
replies                                   4
Name: 233821, dtype: object